sled-agent/src/instance.rs

```diff
@@ -1987,6 +1992,32 @@ impl InstanceRunner {
             opte_ports.push(port);
         }
 
+        // Each delegated Zvol requires delegating the parent dataset
+        let datasets: Vec<_> = self
+            .delegated_zvols
+            .iter()
+            .map(|delegated_zvol| zone::Dataset {
+                name: delegated_zvol.parent_dataset.clone(),
+            })
+            .collect();
+
+        // For delegated devices, include the default list plus any for the
+        // delegated zvol devices.
+        let mut devices = vec![
+            zone::Device { name: "/dev/vmm/*".to_string() },
+            zone::Device { name: "/dev/vmmctl".to_string() },
+            zone::Device { name: "/dev/viona".to_string() },
+        ];
+
+        for delegated_zvol in &self.delegated_zvols {
```

Comment:
nit, take it or leave it: this probably doesn't actually matter, but seems like a nice place for a 
```rust
let mut devices = Vec::with_capacity(self.delegated_zvols.len() + 3);
// ... push the devices
```
to preallocate the necessary size